### DISCUSSION TITLE
在使用 AvPlayer 使用自动播放无法触发`ended` `stopped`事件

### DISCUSSION THREAD

**Discussion by xuanhen2013** [zhaohappy/libmedia#27] [General]
在没有用户进行点击行为时, `audioSource` 看起来是无法通信的
<img width="265" alt="image" src="https://github.com/user-attachments/assets/d930cc23-2b65-4c2a-8c3d-7c227902f1d6">

会导致 [`src/avplayer/AVPlayer.ts#L2254`](https://github.com/zhaohappy/libmedia/blob/master/src/avplayer/AVPlayer.ts#L2254) 中 
```ts
if (this.audioSourceNode) {
      await this.audioSourceNode.request('stop') // 这一步直接无法返回
      this.audioSourceNode.disconnect()
      this.audioSourceNode = null
    }
```
导致后续代码不会执行

大佬看下有什么办法可以避免这个问题吗? 🙏

**Comment by zhaohappy** [comment]
已修复

**Comment by xuanhen2013** [comment]
在自动播放的时候, `audioContext.state = suspended`
调用 `seek` 在
[src/avplayer/AVPlayer.ts#L2012](https://github.com/zhaohappy/libmedia/blob/master/src/avplayer/AVPlayer.ts#L2012) 也会导致后续代码不会执行

需要在点击切换进度条时先执行解锁

**Comment by zhaohappy** [comment]
我测试没有复现，有具体复现步骤吗，测试文件是什么格式的

**Reply by xuanhen2013** [reply]
尝试了很久又没办法稳定复现

复现路径大致为
当状态刚变成 played 的时候 执行 seek

**Reply by zhaohappy** [reply]
没有 await play 就 seek 了吗

**Reply by xuanhen2013** [reply]
应该不是

```
play().then(() => seek(BigInt(10000)))
```
代码执行是没问题的

是通过人为点出来的
具体操作为:
1. 在刚开始播放 (已经出现了渲染界面) 
2. 点击进度条触发 seek (判断状态为 played 才执行)
3. 界面依然在渲染,但是状态变成 seeking

我在大佬你的demo上无法复现这个, 我会后续尝试再复现一下
